hal/src/nRF52840/ble_hal.cpp

```diff
@@ -1004,9 +1003,7 @@ int BleObject::Broadcaster::setAdvertisingData(const uint8_t* buf, size_t len) {
 }
 
 ssize_t BleObject::Broadcaster::getAdvertisingData(uint8_t* buf, size_t len) const {
-    if (buf == nullptr) {
-        return advDataLen_;
-    }
+    CHECK_TRUE(buf, advDataLen_);
```

Comment:
This looks a bit confusing and easy to miss. I personally prefer the original explicit logic.